### DISCUSSION TITLE
Imagebam

### DISCUSSION THREAD

**Discussion by theaulddubliner** [tophf/mpiv#57] [MPIV host rules]
Imagebam seems to have changed its URL. I think /view/ has been added.

https://www.imagebam.com/view/ME2MQG

**Comment by tophf** [comment]
It worked for a while but now the link is 404 so I guess it was a temporary glitch on imagebam.

**Reply by darkred** [reply]
I quote the built-in rule:

https://github.com/tophf/mpiv/blob/a3fefcb0346b9b2474808b93065de0682562447f/script.user.js#L1698-L1703

The given URL example is not covered by the built-in rule:
the /view/ in the given URL example is an *additional* URL pattern to /image/ of the built-in one.


Also, I believe `"q"` needs to be updated.

So, to summarize, If I am not mistaken, the built-in rule should change to:

```
      {
        u: [
			'||imagebam.com/image/',
			'||imagebam.com/view/'
		],
        q: 'img.main-image',
        tabfix: true,
        xhr: hostname.includes('planetsuzy'),
      },
```

**Reply by tophf** [reply]
Imagebam shows "system error" for all images here so I can't even test it. Half an hour ago the built-in rule still correctly worked on `image` links.

**Reply by darkred** [reply]
Imagebam site works again. If you test you'll see that with the built-in rule:

- the OP `view` URL doesn't match  at all,
- and an `image` URL I found, gives `The "q" rule did not produce any URL` :

https://www.imagebam.com/view/ME2MQG
https://www.imagebam.com/image/7fe69728227556

&nbsp;
Both work ok with my suggested change.

**Reply by tophf** [reply]
The built-in rule works with the second link. The first link is still 404 so we should wait until imagebam settles down.

**Reply by darkred** [reply]
I made my tests while keeping Tampermonkey/Violentmonkey background page|`Network` tab open, 
having ticked `Disable cache` checkbox, so it fetches from remote site every time  *(I believe imagebam is working ok now)*,
and I'm getting ` "q" rule did not produce any URL` for the second link consistently.

**Reply by tophf** [reply]
There must be something wrong with the site then because I test with cache disabled as well. In other words nothing should be done for now.

**Comment by theaulddubliner** [comment]
@darkred edit works for me now with the /view/ change.

imagebam is a hopeless host. Slow as treacle when it decides to work.

**Reply by darkred** [reply]
> imagebam is a hopeless host. Slow as treacle when it decides to work.

If that's the case, I guess it would be better to remove it from built-in rules, and just have it in the wiki rules page.
